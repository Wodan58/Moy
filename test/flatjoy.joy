DEFINE
swoncat == swap concat ;
P3 == [ [ [ 1 2 ] [ 3 4 ] [ 5 ] [ ] ] [ [ dup * ] map ] map ] ;
P2 == [ [ 1 2 3 ] [ dup * ] map ] ;
P1 == [ 2 3 + dup * ] ;
P0 == [ ] .
HIDE
j2f-r == [ dup list ] [ [ [ swons ] swoncat ] dip [ j2f-r ] step [ ] swons ]
	 [ [ ] cons [ swoncat ] cons swoncat ] ifte ;
j2f-f == [ dup list ] [ [ [ [ ] ] concat ] dip [ j2f-f ] step [ [ ] cons
	 concat ] concat ] [ [ ] cons [ concat ] cons concat ] ifte ;
IN
j2f-reverse == [ ] swap [ j2f-r ] step [ ] swons ;
j2f-forwards == [ [ ] ] swap [ j2f-f ] step ;
END

P0 .
P0 j2f-forwards .
P0 j2f-forwards i .
P1 .
P1 j2f-forwards .
P1 j2f-forwards i .
P1 j2f-forwards i i .
P2 .
P2 j2f-forwards .
P2 j2f-forwards i .
P2 j2f-forwards i i .
P3 .
P3 j2f-forwards .
P3 j2f-forwards i .
P3 j2f-forwards i i .
P0 .
P0 j2f-reverse .
P0 j2f-reverse i .
P1 .
P1 j2f-reverse .
P1 j2f-reverse i .
P1 j2f-reverse i i .
P2 .
P2 j2f-reverse .
P2 j2f-reverse i .
P2 j2f-reverse i i .
P3 .
P3 j2f-reverse .
P3 j2f-reverse i .
P3 j2f-reverse i i .
P3 j2f-forwards j2f-reverse .
P3 j2f-forwards j2f-reverse i .
P3 j2f-forwards j2f-reverse i i .
P3 j2f-forwards j2f-reverse i i i .
