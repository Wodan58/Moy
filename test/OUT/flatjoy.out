[] 
[[]] 
[] 
[2 3 + dup *] 
[[] [2] concat [3] concat [+] concat [dup] concat [*] concat] 
[2 3 + dup *] 
25 
[[1 2 3] [dup *] map] 
[[] [] [1] concat [2] concat [3] concat [] cons concat [] [dup] concat [*] concat [] cons concat [map] concat] 
[[1 2 3] [dup *] map] 
[1 4 9] 
[[[1 2] [3 4] [5] []] [[dup *] map] map] 
[[] [] [] [1] concat [2] concat [] cons concat [] [3] concat [4] concat [] cons concat [] [5] concat [] cons concat [] [] cons concat [] cons concat [] [] [dup] concat [*] concat [] cons concat [map] concat [] cons concat [map] concat] 
[[[1 2] [3 4] [5] []] [[dup *] map] map] 
[[1 4] [9 16] [25] []] 
[] 
[[]] 
[] 
[2 3 + dup *] 
[[] [*] swoncat [dup] swoncat [+] swoncat [3] swoncat [2] swoncat] 
[2 3 + dup *] 
25 
[[1 2 3] [dup *] map] 
[[] [map] swoncat [] [*] swoncat [dup] swoncat swons [] [3] swoncat [2] swoncat [1] swoncat swons] 
[[1 2 3] [dup *] map] 
[1 4 9] 
[[[1 2] [3 4] [5] []] [[dup *] map] map] 
[[] [map] swoncat [] [map] swoncat [] [*] swoncat [dup] swoncat swons swons [] [] swons [] [5] swoncat swons [] [4] swoncat [3] swoncat swons [] [2] swoncat [1] swoncat swons swons] 
[[[1 2] [3 4] [5] []] [[dup *] map] map] 
[[1 4] [9 16] [25] []] 
[[] [concat] swoncat [] [map] swoncat swons [concat] swoncat [cons] swoncat [] swons [concat] swoncat [] [map] swoncat swons [concat] swoncat [cons] swoncat [] swons [concat] swoncat [] [*] swoncat swons [concat] swoncat [] [dup] swoncat swons [] swons [] swons [concat] swoncat [cons] swoncat [] swons [concat] swoncat [cons] swoncat [] swons [] swons [concat] swoncat [cons] swoncat [] swons [concat] swoncat [] [5] swoncat swons [] swons [concat] swoncat [cons] swoncat [] swons [concat] swoncat [] [4] swoncat swons [concat] swoncat [] [3] swoncat swons [] swons [concat] swoncat [cons] swoncat [] swons [concat] swoncat [] [2] swoncat swons [concat] swoncat [] [1] swoncat swons [] swons [] swons [] swons] 
[[] [] [] [1] concat [2] concat [] cons concat [] [3] concat [4] concat [] cons concat [] [5] concat [] cons concat [] [] cons concat [] cons concat [] [] [dup] concat [*] concat [] cons concat [map] concat [] cons concat [map] concat] 
[[[1 2] [3 4] [5] []] [[dup *] map] map] 
[[1 4] [9 16] [25] []] 
